feat: APP-267 accounts orders page

r41ph opened the pull request:
## Description

https://regennetwork.atlassian.net/browse/APP-267
---

### Author Checklist

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

1. See stories in Storybook: https://deploy-preview-2483--regen-storybook.netlify.app/?path=/story/marketplace-organisms-order--default
2. See the Orders page in the marketplace: https://deploy-preview-2483--regen-marketplace.netlify.app/profile/orders

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

r41ph commented:
@erikalogie see testing instructions

erikalogie commented:
> 1. https://deploy-preview-2483--regen-storybook.netlify.app/?path=/story/marketplace-organisms-order--default

@r41ph you did an awesome job with this! It really looks great, just a few little tiny details.

<img width="655" alt="Monosnap Marketplace : Organisms : Order - Pending Order ⋅ Storybook 2024-09-25 14-43-10" src="https://github.com/user-attachments/assets/3d4ecf00-7276-44a4-a45b-85137a4250da">
<img width="488" alt="Monosnap Marketplace : Organisms : Order - Prefinance ⋅ Storybook 2024-09-25 14-41-37" src="https://github.com/user-attachments/assets/7a10d727-d71b-488c-a3bb-d0dc77468436">
<img width="832" alt="Monosnap Marketplace : Organisms : Order - Tradable Order ⋅ Storybook 2024-09-25 14-49-56" src="https://github.com/user-attachments/assets/c5e85ba5-da3b-4d3b-a1f6-ef6c60eea6c5">
<img width="941" alt="Monosnap Regen Marketplace 2024-09-25 14-46-27" src="https://github.com/user-attachments/assets/d7c9086b-cd3a-480c-857c-b95b81a217f9">

r41ph commented:
> @r41ph you did an awesome job with this! It really looks great, just a few little tiny details.

@erikalogie thanks! Please review the style updates addressing your comments.

erikalogie commented:
@r41ph looks awesome!

One final small thing, and otherwise LGTM!
<img width="444" alt="Monosnap Marketplace : Organisms : Order - Tradable Order ⋅ Storybook 2024-09-26 11-42-06" src="https://github.com/user-attachments/assets/3c905e2a-2587-462e-ac4b-20c540167e5e">

r41ph commented:
@erikalogie have another look when you get a chance please

r41ph commented on web-components/src/components/inputs/new/EditableInput/EditableInput.test.tsx:
```diff
@@ -45,7 +45,7 @@ describe('EditableInput', () => {
 
     const updateButton = screen.getByRole('button', {
-      name: 'update',
+      name: /update/i,
```
To pass the test that was failing after the last translations update. However, if we have a specific reason to enforce case sensitivity an updating 'update' to 'Update' was a mistake, we can revert this and fix the test by changing the buttons label casing.

r41ph commented on web-marketplace/src/components/organisms/Order/Order.Credits.tsx:
```diff
@@ -0,0 +1,66 @@
+  const { credits, price, denom } = data as CreditsData;
+  const totalPrice = +credits * +price;
+  return (
+    <section className="flex gap-20">
```
In this case, I think creating a reusable component for these sections may add unnecessary complexity because the logic for each component is slightly different. While they share some structure (title, icon, rows), combining them into a single component would require a few conditional checks that would make them harder to work with. I believe that keeping them separated while using shared internal components allows each component to be simple to maintain and update.

blushi replied:
then what about just having a reusable component with title, icon and children, which makes it more flexible?

r41ph commented on web-marketplace/src/components/organisms/Order/Order.MakeAnonymous.tsx:
```diff
@@ -0,0 +1,71 @@
+            name={name}
+            autoFocus
+          >
+            <option value="Yes">Yes</option>
+            <option value="No">No</option>
```
I have added the translations although I wasn't sure because we don't need them now because this component is not going to be fully implemented at this stage.

blushi replied:
I believe that as long as we create new components, we should add translations everywhere we should, even though it's not complete yet, because even if we'll work again on this component, we might just forget about the texts to translate

r41ph commented on web-marketplace/src/clients/regen/Regen.Routes.tsx:
```diff
@@ -337,6 +339,7 @@ export const getRegenRoutes = ({
             element={<AuthRoute component={ProfileEditSettings} />}
           />
         </Route>
+        <Route path="orders" element={<AuthRoute component={Orders} />} />
```
I have created a task for that: https://regennetwork.atlassian.net/browse/APP-375?atlOrigin=eyJpIjoiMTJlMTlhMGQ1YmRkNDUxZGE5ODAxODQ1ZjE0Yzk1ZTgiLCJwIjoiamlyYS1zbGFjay1pbnQifQ

blushi replied:
ok cool thanks
please don't forget to put the "dev" label when you create new tasks

r41ph commented on web-marketplace/src/pages/Orders/Orders.tsx:
```diff
@@ -0,0 +1,109 @@
+import { Order } from 'components/organisms/Order/Order';
+import { ORDER_STATUS } from 'components/organisms/Order/Order.constants';
+import {
+  blockchainDetails,
+  credits,
+  paymentInfo,
+  retirementInfo,
```
I didn't translate them because they are mocks and we may end up creating them in a different way, perhaps inside the component with the real data I thought.

blushi replied:
yeah but the titles aren't going to change, they are not dynamic data, so we could put them directly in the components

blushi replied:
in the dashboard case, this is only static info that we extracted so it can be reusable (in the menu and on the dashboard)
Here it's a bit different, I just find a bit weird to mix up static info with dynamic content.

r41ph commented on web-marketplace/src/components/organisms/Order/Order.tsx:
```diff
@@ -0,0 +1,144 @@
+    (retirementInfo.data as RetirementInfoData).tradableCredits !== null;
+  return (
+    <Card className="bg-grey-100 border-[1px] border-grey-200">
+      {isMobile && (
```
I think is better to not render them instead of just hide them as hidden elements still exist in the DOM. Also, we use `useMediaQuery` in many places already and thought we'll probably end up needing to create a `useMediaQuery` hook if we migrate from MUI.

erikalogie commented on tailwind.css:
```diff
@@ -48,6 +48,8 @@
     --purple-200: 187 195 210;
     --purple-100: 221 225 233;
 
+    --orange-700: 122 64 11;
```
Not yet, I would have to change all the colors in the fiat comps to be the newly created contextual colors

erikalogie commented:
@r41ph looks great except now the left-alignment got messed up a bit: 
<img width="476" alt="Monosnap Marketplace : Organisms : Order - Crypto ⋅ Storybook 2024-09-30 08-13-37" src="https://github.com/user-attachments/assets/29b02e94-7960-4694-9509-4a8877f82337">

r41ph commented:
@erikalogie please have another look

erikalogie commented:
so nice! LGTM

blushi reviewed (commented):
I think the font for those labels is supposed to be Muli, not Lato, see https://www.figma.com/design/BTuUv6QXY4GbliZcXCe8RJ/Fiat-payments?node-id=3234-100679&node-type=instance&m=dev

![image](https://github.com/user-attachments/assets/eef34b62-cef0-4b9e-9b5b-683046d15d54)

I believe we could potentially reuse our [`TextButton` component](https://regen-storybook.netlify.app/?path=/story/buttons--text-button) (getting rid of the hovering changing color) and this is how it's defined in figma as well

blushi commented on web-marketplace/src/components/organisms/Order/Order.Summary.tsx:
```diff
@@ -0,0 +1,189 @@
+              </div>
+            }
+            value={
+              <Link
+                href="https://google.com"
+                className="text-grey-700 !underline inline-block text-ellipsis w-[100px] overflow-hidden"
+              >
+                {blockchainRecord}
+              </Link>
```
we should use the same logic as in https://github.com/regen-network/regen-web/blob/d016f292e058c61b73f9947c80f7fb8184c79a8c/web-marketplace/src/components/organisms/ConfirmationModal.tsx#L134-L144

blushi commented on web-marketplace/src/components/organisms/Order/Order.Summary.tsx:
```diff
@@ -0,0 +1,189 @@
+  const { purchaseDate, blockchainRecord } =
+    blockchainDetails as BlockchainDetailsData;
+  const { nameOnCard, askDenom, cardLast4 } = paymentInfo as PaymentInfoData;
+  const isCardPayment = nameOnCard && cardLast4 && askDenom === 'usd';
```
```suggestion
  const isCardPayment = nameOnCard && cardLast4 && askDenom === USD_DENOM;
```

blushi commented on web-marketplace/src/components/organisms/Order/Order.MakeAnonymous.tsx:
```diff
@@ -0,0 +1,80 @@
+            name={name}
+            autoFocus
+          >
+            <option value={i18n._(msg`Yes`)}>
```
An option value does not need to be internationalized, only the displayed label

blushi commented on web-marketplace/src/components/organisms/Order/Order.Summary.tsx:
```diff
@@ -0,0 +1,189 @@
+                currencyCode={creditAskDenom}
+                className="mr-10 text-base"
+              />
+              <DenomIconWithCurrency
```
please check how I updated it in my PR: https://github.com/regen-network/regen-web/pull/2477/files#diff-6e92c61d1ad4141c61e04825b5cd429147b6a13e6e902f0d95d415002909384c
we need ask**Base**Denom and displayDenom

r41ph replied:
I've updated this to follow what you do in your PR. Please let me know if any further changes are required.

blushi commented on web-marketplace/src/components/organisms/Order/Order.SummaryRow.tsx:
```diff
@@ -0,0 +1,38 @@
+      <Title
+        variant="h6"
+        className={cn(
+          "text-grey-400 text-[11px] font-extrabold font-['muli'] uppercase tracking-[1px] mt-[3px]",
```
```suggestion
          "text-grey-400 text-[11px] font-extrabold uppercase tracking-[1px] mt-[3px]",
```

blushi commented on web-marketplace/src/components/organisms/Order/Order.Summary.tsx:
```diff
@@ -0,0 +1,189 @@
+              title={_(msg`card info`)}
+              value={
+                <span>
+                  {_(msg`Visa ending in`)} {cardLast4}
```
this is not necessarily a visa, the card brand should come as a prop

blushi commented on web-marketplace/src/components/organisms/Order/Order.types.tsx:
```diff
@@ -0,0 +1,58 @@
+import { Currency } from 'web-components/src/components/DenomIconWithCurrency/DenomIconWithCurrency.constants';
```
```suggestion
```

blushi commented on web-marketplace/src/components/organisms/Order/Order.Summary.tsx:
```diff
@@ -0,0 +1,189 @@
+} from './Order.types';
+
+interface OrderSummaryProps {
+  retirementInfo: OrderSummarySectionProps;
+  blockchainDetails: OrderSummarySectionProps;
+  creditsData: OrderSummarySectionProps;
+  paymentInfo: OrderSummarySectionProps;
```
why setting them all as `OrderSummarySectionProps`?

r41ph replied:
Since the update of the `title` and `icon` as props this is doesn't need to be like this, thanks

clevinson commented:
Nice work! just QA'd the flows and looks good to me!

One small on text is that I think it looks better if it says "Credit issuance pending" instead of "Credits issuance pending".
<img width="421" alt="Screenshot 2024-10-02 at 4 51 01 PM" src="https://github.com/user-attachments/assets/ee9f8f48-07d3-4e94-89b9-740aac72acea">

r41ph commented:
> One small on text is that I think it looks better if it says "Credit issuance pending" instead of "Credits issuance pending". 
And the other one: "Credits delivered", is this alright in plural?

erikalogie commented:
> > One small on text is that I think it looks better if it says "Credit issuance pending" instead of "Credits issuance pending".
> 
> And the other one: "Credits delivered", is this alright in plural?

Yes this is good

blushi reviewed (approved):
LGTM pre-approving
Pending small UI issue to be fixed for the transaction hash, the ellipsis is not underlined as it's shown on figma:

![image](https://github.com/user-attachments/assets/063d7085-1b5c-428c-aa1d-b91b2416cdfd)

blushi commented on web-components/src/components/cards/OrderSummaryCard/OrderSummaryCard.test.tsx:
```diff
@@ -49,23 +49,25 @@ describe('OrderSummaryCard', () => {
       name: 'editable-credits',
     });
-    fireEvent.change(editInput, { target: { value: 7 } });
+    if (editInput) {
```
looks like `editInput` is always null

r41ph replied:
nice catch, thanks
